### PR TITLE
AI can no longer hack airlocks when it's not rogue/malf

### PR DESCRIPTION
**Pull request by ShiftyRail**
[Balance] [Server vote] [etc]

### Reasoning

Why should the AI control wire be just a very small delay to the AI control of the systems instead of a more permanent way to deal with the big silicon overlord ?
Sure, you can disable cameras, but it is obvious that it instantly gives you away and is a sure way to have 99 cyborgs converging towards you.

This does not apply to malf/traitor AIs, who can still hack airlocks to their heart's content.
Alternative solutions : 
- Longer timer for the hacking
- Chances of failing to hack
- Only a short window of time to act when you hack into the airlock.

:cl:
- rscdel: Airlock hacking has been restricted to only rogue/malf AIs, so that if you disable an AI control wire, it's a reliable way to deal with the AI instead of a slight delay.

### CODE REVIEW

**Comment by Reviire** [comment]
This is stupid. Time to enforce the plasma flood meta.

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -630,6 +630,11 @@ About the new airlock wires panel:
 
 
 /obj/machinery/door/airlock/proc/hack(mob/user as mob)
+
+	if (!(isAI(user) && user.mind && user.mind.special_role))
+		to_chat(user, "Airlock AI control has been blocked. Dispatch a cyborg, or a carbon engineer, for maintenance.")
```
Comment (boy2mantwicethefam) [inline code/game/machinery/doors/airlock.dm:634]: engineers might also be androids, plasmen, vox, and some other stuff. Replace carbon with organic cuz that's racist and organophobe

**Comment by boy2mantwicethefam** [comment]
This is ok but the hacking duration should be much shorter if it's malf instead of the default duration

**Comment by ShiftyRail** [comment]
This is about a "friendly" AI having control when it should not have, ie when you try to remove its access from, say, perma doors, or when you do not want it to permanently bolt you in.

This is not about antag AIs, since they are specifically exempt from that change.

**Comment by boy2mantwicethefam** [comment]
I know the purpose of the PR is to prevent AIs from having access to the airlock if the control wire was cut but I'm suggestin malfs should hack the door faster cuz it's like 49 seconds right now at least according to the code

File: code/game/machinery/doors/airlock.dm (hunk 1 above)
Comment (SonixApache) [reply code/game/machinery/doors/airlock.dm:634]: did you know that plasmemes and birdniggers are carbons

**Comment by Pathid** [comment]
Doesn't this enable a type of antag-checking? 

>order AI to open door you cut the wire on
>antag AI clicks door
>automatically hacks door since there's no option not to
>later, order AI to open door again
>AI doesn't have a way to know if the wire is mended or not

**Comment by DamianX** [comment]
nice catch
I suggest we either remove silicons or remove the ability for them to interact with airlocks with the cut wire at all

**Comment by Pathid** [comment]
What I suggest is:
* Only hacking one door at a time
* Hacking a second door unhacks the first door
* Longer timer
* Antag AIs have an option to put the process into overdrive (basically what it is now) at their discretion

**Comment by ShiftyRail** [comment]
@Pathid I'll give malf AIs an option rather than automatic hacking to deal with that.
As a bonus antag AIs will be able to do it faster

**Comment by SonixApache** [comment]
re-implement tetris and make door hacking that

**Comment by Reviire** [comment]
There's still no reason to nerf this, when you already have an option to completely remove the AI from the area. You can cut cameras. It's extremely obvious, but that's the catch. You either be extremely obvious and entirely remove the AI from play, or you be very subtle and if the AI does eventually notice, it can spend a whole minute gaining control over a single door.

All you're doing is giving them what is essentially the power of cutting a camera, for the ease and subtlety of cutting a wire the AI can't see. What if instead, we started encouraging antags to use their brains, the things you can do with a signaler are amazing.

**Comment by Pathid** [comment]
@Reviire I do have to wonder why doors are now a problem, when they already include a whole variety of ways to screw over the AI:
* Cutting the power wires along with the AI control wire locks the AI completely out of the door, with the caveat that it's now an unpowered door you have to crowbar to open and close. (If sec wants to secure perma against the AI all they'd have to do is do this on one set of the double perma doors.)
* Cutting the bolt wire: This is a popular way to avoid getting doorcrushed since it's only a single wire once the door is open and there's not a thing the AI can do to unbolt it without an engiborg.
* Cutting the electrify wire: permanently shocks the door and again, there's nothing the AI can do directly. Though it can briefly cut the power to that door or alternatively cut the power to the whole area through the APC (unless the AI control wire on that is cut, and in that case, tough potatoes.)

I'm not even getting into signalers, which like the above the AI has no way to remove without an engiborg.

**Comment by Blithering** [comment]
I do think AIs should have to rely on their cyborgs for things like this, being able to fix cut wires remotely is a bit odd

**Comment by Pathid** [comment]
Physically, the wire is still cut. Hacking doesn't allow cyborgs to operate it remotely for what that's worth. One way to look at it is, airlocks are the AI's main physical presence on the station. It means there's multiple approaches to taking that basic thing away with trade-offs to them (bolt wire, power wires, cameras, APCs), instead of the reductive "cut this single wire and the rest doesn't matter, you're done."

This needs an [Update Wiki] tag by the way.
http://ss13.moe/wiki/index.php/Hacking#Airlocks

**Comment by ShiftyRail** [comment]
Proof that this works : 
proof

**Comment by Pathid** [comment]
Even with that, I don't like this. It's taking complexity away from a system (messing with door wires) that has a reason to be complex, making it take no strategy to do something you can already do, and having the exception only apply to times when the two AI antag roles go loud. (One of them badly in need of a rework and the other probably needing to be removed.) There's a limited number of wire-related "hacks" for a player to figure out, and working out the real ways to lock out the AI is one of them.

**Comment by SonixApache** [comment]
re-add door alarms

**Comment by ShiftyRail** [comment]
This will need to be re-done due to role datums but fear not

File: vgstation13.dme [hunk 2]
```diff
@@ -2414,8 +2414,8 @@
 #include "interface\web\info.dms"
 #include "interface\web\interface.dms"
 #include "maprendering\maprendering.dm"
-#include "maps\_map_override.dm"
 #include "maps\_map.dm"
+#include "maps\_map_override.dm"
```
Comment (ShiftyRail) [inline vgstation13.dme:2418]: lmao @DamianX, it somehow changed places with the macros.
I don't really know how & why it happens, or if I did a mistake.

**Comment by ShiftyRail** [comment]
This is now role-datums compliant.

File: vgstation13.dme (hunk 2 above)
Comment (DamianX) [reply vgstation13.dme:2418]: I included that file manually and apparently got it wrong, I'm okay with this fixing it

**Comment by SonixApache** [comment]
why would you merge an antagcheck

**Comment by DamianX** [comment]
Feel free to open a revert

**Pull request merged**